src/shared/utilities/fileSystemWatcher.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+/**
+ * Interface representing an object that can listen to a vscode.FileSystemWatcher
+ */
+export interface FileWatcherListener extends vscode.Disposable {
+    onListenedChange(templatePath: vscode.Uri): Promise<void>
+    onListenedCreate(templatePath: vscode.Uri): Promise<void>
+    onListenedDelete(templtemplatePathateUri: vscode.Uri): Promise<void>
+}
+
+/**
+ * Creates a file watcher that attaches functions to onDidChange, onDidCreate, and onDidDelete functions
+ *
+ * The resulting file watcher will check the glob pattern across all workspace folders.
+ * @param listener FileWatcherListener that includes functions to attach to onEvent hooks
+ * @param globPattern Glob pattern to use for file watcher
+ */
+export function associateFileSystemWatcherWithListener(
```

Comment:
"associate" is a rather ceremonious verb. Is there a reason for the unconventional name?  A hashmap is an "associative array" yet its methods are still just `set` and `get`. 

Looks like this is doing "setup" so this method could be named `setupFSWatcherListener()`. Much shorter and without any loss of clarity. I won't comment on other similar cases, but hope that this will be taken as general comment on other cases, and in the future.

BTW: "fs" is a vastly common name for "file system". Let's not spell out "file system" everywhere, just like we don't spell out "Hypertext Markup Language" instead of HTML.

Reply:
I'm with you on replacing `associate`, but I'm not so sure about the others. I'd be interested to know what you'd want to replace `listener` with, since that would be pervasive throughout the rest of the PR. I don't like the idea of swapping out `FileSystemWatcher` for `FS` due to the fact that this specifically works with the `vscode.FileSystemWatcher` construct; this is not a generic Node `fs.watch`.

Reply:
Its plausible that we make separate filesystem watcher instances for different globs in the future. We'd want to associate them with a listener. I think this method is better moved to a public `addFileSystemWatcher` method on the listener class.